src/cli/generator/generator.rs

```diff
@@ -180,13 +180,7 @@ impl Generator {
         let mut config = config_gen.mutation(mutation_type_name).generate(true)?;
 
         if infer_type_names {
-            let key = if !secret.is_empty() {
-                Some(secret.to_string())
-            } else {
-                None
-            };
-
-            let mut llm_gen = InferTypeName::new(key);
+            let mut llm_gen = InferTypeName::new(llm_config);
```

Comment:
```suggestion
            let mut llm_gen = InferTypeName::new(model, key);
```

Pass the model and the key directly from here so that we don't need to verify if the input is valid.